### PR TITLE
MAKE-887: Set bucket, path & createdAt default values on test artifacts

### PR DESCRIPTION
**Pull request by makesitgo**

### CODE REVIEW

**Comment by makesitgo** [comment]
@tychoish @julianedwards ... if you don't mind taking a look here!  I'm not sure if my changes to `report_artifact_test.go` were the right direction or not

File: report_artifact_test.go [hunk 1]
```diff
@@ -147,7 +147,7 @@ func TestUpload(t *testing.T) {
 			bucketConf: BucketConfiguration{
 				Region: s3Region,
 			},
-			hasErr: true,
+			dryRunNoErr: true,
```
Comment (julianedwards) [inline report_artifact_test.go]: You can remove this line, `dryRunNoError` only needs to be set when `hasErr` is set to true

File: report_artifact_test.go [hunk 2]
```diff
@@ -181,6 +181,18 @@ func TestUpload(t *testing.T) {
...
+			name: "BucketSpecifiedFromConfiguration",
+			artifact: &TestArtifact{
+				Path:      "bson_example.bson",
+				LocalFile: "testdata/bson_example.bson",
+			},
+			bucketConf: BucketConfiguration{
+				Name:   "bucket",
+				Region: s3Region,
+			},
+			dryRunNoErr: true,
```
Comment (julianedwards) [inline report_artifact_test.go]: You can remove this one as well

File: report_artifact_test.go (hunk 1 above)
Comment (julianedwards) [reply report_artifact_test.go]: Also, to make tests pass change line 144 to `Bucket: s3Name`

File: report_artifact_test.go [hunk 3]
```diff
@@ -181,6 +181,18 @@ func TestUpload(t *testing.T) {
...
 			hasErr: true,
 		},
+		{
+			name: "BucketSpecifiedFromConfiguration",
+			artifact: &TestArtifact{
+				Path:      "bson_example.bson",
+				LocalFile: "testdata/bson_example.bson",
+			},
+			bucketConf: BucketConfiguration{
+				Name:   "bucket",
```
Comment (julianedwards) [inline report_artifact_test.go]: `Name: "s3Name",`

**Comment by makesitgo** [comment]
@julianedwards ... thanks for the initial pass, requested changes have been pushed up

File: report_artifact_test.go [hunk 4]
```diff
@@ -141,13 +141,12 @@ func TestUpload(t *testing.T) {
 		{
 			name: "NoRemotePath",
 			artifact: &TestArtifact{
-				Bucket:    "bucket",
+				Bucket:    "s3Name",
```
Comment (julianedwards) [inline report_artifact_test.go]: Sorry, didn't specify clearly but this is a variable `s3Name` so no need for the quotation marks.

File: report_artifact_test.go [hunk 5]
```diff
@@ -181,6 +180,17 @@ func TestUpload(t *testing.T) {
...
 			hasErr: true,
 		},
+		{
+			name: "BucketSpecifiedFromConfiguration",
+			artifact: &TestArtifact{
+				Path:      "bson_example.bson",
+				LocalFile: "testdata/bson_example.bson",
+			},
+			bucketConf: BucketConfiguration{
+				Name:   "s3Name",
```
Comment (julianedwards) [inline report_artifact_test.go]: Same here

File: report_artifact_test.go (hunk 4 above)
Comment (makesitgo) [reply report_artifact_test.go]: hahaha I absolutely should have caught on there...

:bushes:

**Pull request merged**